### PR TITLE
perf(meta): make GlobalStreamManager::build_actors concurrent

### PR DESCRIPTION
**Pull request by TennyZhuang**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Found by #14013 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by yezizp2012 — COMMENTED** [review]
LGTM, could you also please update the same logic in recovery?

File: src/meta/src/barrier/recovery.rs [hunk 1]
```diff
@@ -638,18 +640,28 @@ impl GlobalBarrierManager {
             return Ok(());
         }
```
Comment (yezizp2012) [inline src/meta/src/barrier/recovery.rs:642]: And the function `update_actors` as well?

**Review by yezizp2012 — APPROVED** [review]
Rest LGTM

File: src/meta/src/stream/stream_manager.rs [hunk 2]
```diff
@@ -395,45 +397,65 @@ impl GlobalStreamManager {
...
+                client
+                    .update_actors(UpdateActorsRequest {
+                        request_id,
+                        actors: stream_actors.clone(),
+                    })
+                    .await?;
+
+                Ok::<_, MetaError>(())
+            }
+        }).collect::<FuturesUnordered<_>>();
```
Comment (stdrc) [inline src/meta/src/stream/stream_manager.rs:427]: Is there a result buffer size for this? Or we just use it and don't care about how it stores the buffered result, if any?

The description of this type says it is **optimized to manage a large number of futures**, if true, seems we can also use it in our HashAgg implementation.

**Review by stdrc — COMMENTED** [review]
Generally LGTM

File: src/meta/src/stream/stream_manager.rs (hunk 2 above)
Comment (TennyZhuang) [reply src/meta/src/stream/stream_manager.rs:427]: May combined with buffer_unordered?

**Review by BugenZhao — APPROVED** [review]

**Review by yezizp2012 — APPROVED** [review]

**Pull request merged**